depscan/lib/utils.py

```diff
@@ -257,9 +257,21 @@ def search_pkgs(project_type: str | None, pkg_list: List[Dict[str, Any]]):
                     purl_aliases[f"{vari_full_pkg.lower()}:{version}"] = pkg.get("purl")
     raw_results = []
     for i in expanded_list:
-        search_term = i.get("purl") or i.get("name")
-        if res := search_by_purl_like(search_term, with_data=True):
-            raw_results.extend(res)
+        if i.get("purl"):
+            if res := search_by_purl_like(i.get("purl"), with_data=True):
+                raw_results.extend(res)
+        elif i.get("cpe"):
+            if res := search_by_cpe_like(i.get("cpe"), with_data=True):
+                raw_results.extend(res)
+        elif i.get("url"):
+            if res := search_by_url(i.get("url"), with_data=True):
+                raw_results.extend(res)
+        else:
+            alt_search_term = f"pkg:generic/{i.get('vendor')}/{i.get('name')}" if i.get("vendor") else i.get("name")
+            if i.get("version"):
+                alt_search_term = f"{alt_search_term}@{i.get('version')}"
+            if res := search_by_purl_like(alt_search_term, with_data=True):
```

Comment:
@prabhu This statement needs to be nested under the previous or alt_search_term might be undefined.